src/kafel.c

```diff
@@ -87,14 +74,46 @@ KAFEL_API void kafel_set_input_file(kafel_ctxt_t ctxt, FILE* file) {
   ASSERT(ctxt != NULL);
   ASSERT(file != NULL);
 
+  char *filebuf = NULL;
+
+  // Free and null the string if the type was FILE
+  if (ctxt->input.string && ctxt->input.type == INPUT_FILE) {
+    char *freestr = (char*)ctxt->input.string;
```

Comment:
I assumed the previous type was set to FILE so `kafel_set_input_file` allocated the string for us. This can be simplified if we use the approach with `kafel_copy_ctxt` you suggested, though - we will need no caching the policy file contents anymore.